fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -631,8 +636,11 @@ private TopDocs getPageOfData(
             int perPage,
             int currentPage
     ) {
+        RequestLog.startTiming("QueryingLuceneForPage" + currentPage + "ReadLock");
         lock.readLock().lock();
+        RequestLog.stopTiming("QueryingLuceneForPage" + currentPage + "ReadLock");
         try {
+            RequestLog.startTiming("QueryingLuceneForPage" + currentPage);
```

Comment:
If there are many pages, that could create a large bloat to the request log. I would move this timing up a level. You can calculate the per-page timings using the pagination information that I _think_ should already be in the RequestLog.